pkg/plugin/plugin.go

```diff
@@ -0,0 +1,92 @@
+package plugin
+
+import (
+	"os/exec"
+	"path/filepath"
+
+	"github.com/dcos/dcos-cli/pkg/cli"
+	"github.com/spf13/cobra"
+)
+
+// Plugin defines an external plugin and its associated data.
+type Plugin struct {
+	Name        string        `yaml:"name"`
+	Description string        `yaml:"description"`
+	Version     string        `yaml:"version"`
+	Source      source        `yaml:"source"`
+	Executables []*executable `yaml:"executables"`
+
+	// Plugin directory in filesystem
+	dir string
+
+	// directory containing the plugin's binaries
+	binDir string
+}
+
+// executable defines what commands are associated with which executable file in the plugin.
+type executable struct {
+	// Executables are found in the binary directory + the filename.
+	// This means all executables in a plugin are in the same place.
+	Filename string     `yaml:"filename"`
+	Commands []*command `yaml:"commands"`
+}
+
+// command is a command living within a plugin binary.
+type command struct {
+	Name        string `yaml:"name"`
+	Description string `yaml:"description"`
+}
+
+// source is the type and address of the plugin.
+type source struct {
+	Type string `yaml:"type"`
+	URL  string `yaml:"url"`
+}
+
+// IntoCommands creates a list of cobra commands from the commands available in the plugin
+func (p *Plugin) IntoCommands(ctx *cli.Context) []*cobra.Command {
+	var commands []*cobra.Command
+
+	for _, e := range p.Executables {
+		for _, c := range e.Commands {
+			cmd := c.intoCobra(ctx, p.binDir, e.Filename)
+
+			commands = append(commands, cmd)
+		}
+	}
+
+	return commands
+}
+
+// intoCobra creates a Cobra command used to call this command.
+func (c *command) intoCobra(ctx *cli.Context, dir string, exe string) *cobra.Command {
```

Comment:
It'd be preferable not to introduce dependencies on cobra in our packages outside of `cmd`.